pkg/migrations/duplicate.go

```diff
@@ -134,12 +133,16 @@ func (d *Duplicator) Duplicate(ctx context.Context) error {
 		}
 	}
 
-	// Generate SQL to duplicate any unique constraints on the columns
-	// The constraint is duplicated by adding a unique index on the column concurrently.
+	// Create indexes for unique constraints on the columns concurrently.
 	// The index is converted into a unique constraint on migration completion.
-	for _, sql := range d.stmtBuilder.duplicateUniqueConstraints(d.withoutConstraint, colNames...) {
-		if _, err := d.conn.ExecContext(ctx, sql); err != nil {
-			return err
+	for _, uc := range d.stmtBuilder.table.UniqueConstraints {
+		if slices.Contains(d.withoutConstraint, uc.Name) {
+			continue
+		}
+		if duplicatedMember, constraintColumns := d.stmtBuilder.allConstraintColumns(uc.Columns, colNames...); duplicatedMember {
+			if err := createUniqueIndexConcurrently(ctx, d.conn, "", DuplicationName(uc.Name), d.stmtBuilder.table.Name, constraintColumns); err != nil {
```

Comment:
Is passing `""` as the schema name here safe? 

What happens if we are duplicating a unique constraint on a column on a table in a non-`public` schema?

Reply:
It is as safe as it was before this PR. We were already generating create index statements using `pq.QuoteIdentifier(d.table.Name)`, without schema name. No change in behavior. 

Though, I'd prefer to always have schema name, using qualified names instead of just object names. I was not able to do it here because at this point of the code we don't have it. We should maybe consider adding schema name info into `Duplicator` or `duplicatorStmtBuilder`.

Reply:
https://github.com/xataio/pgroll/blob/main/pkg/migrations/duplicate.go#L208

For instance, we don't use schema name here as well. What happens if the table is in a non-public schema?

Reply:
Ok, it works because `pgroll` sets the Postgres `search_path` to whatever the `--schema` flag is set to:

https://github.com/xataio/pgroll/blob/c09619aa879e0a0e8782eb495962cfe8a800a559/pkg/roll/roll.go#L100-L101

So there is no need to explicitly schema qualify names here at all.